cpp/cmake/Dependencies.cmake

```diff
@@ -126,17 +126,14 @@ set(TREELITE_DIR ${CMAKE_CURRENT_BINARY_DIR}/treelite CACHE STRING
   "Path to treelite install directory")
 ExternalProject_Add(treelite
     GIT_REPOSITORY    https://github.com/dmlc/treelite.git
-    GIT_TAG           6fd01e4f1890950bbcf9b124da24e886751bffe6
+    GIT_TAG           e776e202fa732816e0a617715250fc727d573edf
```

Comment:
Next-gen Treelite: https://github.com/dmlc/treelite/tree/treelite-v1
* Flat Tree representation
* Python protocol inferface for the model object, to enable zero-copy serialization
* Revamped build system, with Modern CMake convention